monai/losses/nacl_loss.py

```diff
@@ -0,0 +1,279 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import math
+from typing import Any
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.modules.loss import _Loss
+
+from monai.utils import pytorch_after
+
+
+def get_mean_kernel_2d(ksize: int = 3) -> torch.Tensor:
+    mean_kernel = torch.ones([ksize, ksize]) / (ksize**2)
+    return mean_kernel
+
+
+def get_mean_kernel_3d(ksize: int = 3) -> torch.Tensor:
```

Comment:
Perhaps we can combine these two to one function and move it under this [folder](https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/layers/convutils.py) to make it to be a more common used utils. What do you think?

cc @ericspod @Nic-Ma

Reply:
```python
def get_mean_kernel(ksize: int = 3, dim: int = 2) -> torch.Tensor:
    return torch.ones([ksize] * dim) / (ksize**dim)
```
Something like this.

Reply:
Sure, got the idea. Will work on this as well as for the gaussian (https://github.com/Project-MONAI/MONAI/pull/7819#discussion_r1705520061)